File: DungeonCrawlers/Entities/Character.cs
using DungeonCrawlers.Components;

namespace DungeonCrawlers.Entities
{
    public class Character : ICharacter
    {
        public Character(IName name, IRace race, IHealth health, IArmour armour, IWeapon weapon)
        {
            Name = name;
            Race = race;
            Health = health;
            Armour = armour;
            Weapon = weapon;
        }

        public IName Name { get; }
        public IRace Race { get; }
        public IHealth Health { get; }
        public IArmour Armour { get; }
        public IWeapon Weapon { get; }
    }
}

File: models/character.go
package models

type Character struct {
	Name   string
	Class  string
	Weapon Weapon
	Armour Armour
}

func createCharacter(name string, class string) Character {
	return Character{
		Name:   name,
		Class:  class,
		Weapon: createWeapon(5, 10),
		Armour: createArmour(100, 100),
	}
}
